### PR TITLE
ARM support and build system refactoring

### PR DESCRIPTION
**Pull request by rafie**
- Dockerfile.arm for ARM (arm64v8/arm32v8) cross-build
- /build/docker/Makefile for multi-arch Docker image construction
- Updated /deps/readies

### CODE REVIEW

File: deps/readies/LICENSE [hunk 1]
```diff
@@ -1,52 +1,29 @@
...
+BSD 3-Clause License
```
Comment (MeirShpilraien) [inline deps/readies/LICENSE]: So you are changing the license :) ?

File: deps/readies/bin/getredis5 [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(self, nop=False):
+        paella.Setup.__init__(self, nop)
+
+    def common_first(self):
+		pass
+
+    def debian_compat(self):
+		# https://chilts.org/installing-redis-from-chris-leas-ppa/
+        self.run("add-apt-repository -y ppa:chris-lea/redis-server")
+		self.install("redis-server")
```
Comment (MeirShpilraien) [inline deps/readies/bin/getredis5]: You have extra tabs here, might cause issues in python ...

File: deps/readies/bin/getredis5 [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    def redhat_compat(self):
+		# https://linuxize.com/post/how-to-install-and-configure-redis-on-centos-7/
+        self.install("epel-release yum-utils")
+
+        self.install("http://rpms.remirepo.net/enterprise/remi-release-7.rpm")
+		self.run("yum-config-manager -y --enable remi")
```
Comment (MeirShpilraien) [inline deps/readies/bin/getredis5]: same

File: deps/readies/bin/getredis5 [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		self.install("redis")
+
+    def fedora(self):
+		self.install("dnf-plugins-core")
```
Comment (MeirShpilraien) [inline deps/readies/bin/getredis5]: same

File: deps/readies/LICENSE (hunk 1 above)
Comment (rafie) [reply deps/readies/LICENSE]: RLTest is BSD, most of stuff in RedisLabsModules is Apache, MIT, BSD.

File: deps/readies/bin/getredis5 (hunk 2 above)
Comment (rafie) [reply deps/readies/bin/getredis5]: Fixed.

File: deps/readies/bin/getredis5 (hunk 4 above)
Comment (rafie) [reply deps/readies/bin/getredis5]: Fixed.

File: deps/readies/bin/getredis5 (hunk 3 above)
Comment (rafie) [reply deps/readies/bin/getredis5]: Fixed.

File: deps/readies/LICENSE (hunk 1 above)
Comment (rafie) [reply deps/readies/LICENSE]: BTW, if readies is virtually a dependent module (say, like libevent), but contained in Gears verbatim, can it maintain a separate license?

File: Dockerfile [hunk 5]
```diff
@@ -1,24 +1,32 @@
...
-RUN set -ex; \
-    apt-get update; \
-	apt-get install -y python git
+ADD . /build
+WORKDIR /build
 
-ADD . /redisgears
-WORKDIR /redisgears
+RUN ./deps/readies/bin/getpy2
+RUN python ./system-setup.py
```
Comment (MeirShpilraien) [inline Dockerfile]: write a comment that we may need to remove the python command

Comment (rafie) [reply Dockerfile]: Removed "python".

**Pull request merged**